### PR TITLE
Adds Remote Playback API requirements.

### PR DESCRIPTION
**Pull request by mfoltzgoogle**
Addresses Issue #3: Requirements for the Remote Playback API.

These are derived from the Remote Playback Use Cases & Requirements doc [1] and the Remote Playback spec itself [2], specifically some of the implementation-specific steps and the non-normative notes for implementers.

[1] https://github.com/w3c/remote-playback/blob/gh-pages/use-cases.md
[2] https://w3c.github.io/remote-playback/

PTAL @pthatcherg

***
<a href="https://pr-preview.s3.amazonaws.com/webscreens/openscreenprotocol/pull/162.html" title="Last updated on May 22, 2019, 1:54 AM UTC (6f13996)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webscreens/openscreenprotocol/162/fb68050...6f13996.html" title="Last updated on May 22, 2019, 1:54 AM UTC (6f13996)">Diff</a>

### CODE REVIEW

**Comment by mfoltzgoogle** [comment]
Also, @mounirlamouri may want to look as well, as it relates to Remote Playback API.

File: index.bs [hunk 1]
```diff
@@ -187,7 +188,27 @@ Presentation API Requirements {#requirements-presentation-api}
 Remote Playback API Requirements {#requirements-remote-playback}
 ----------------------------------------------------------------
 
-Issue(3): Requirements for Remote Playback API.
+1.  A [=controlling user agent=] must be able to find out whether there is at
+    least one compatible [=remote playback device=] available for a given
+    {{HTMLMediaElement}}, both instantaneously and continuously.
+
+2.  A controlling user agent must be able to start sending the content of an
```
Comment (mounirlamouri) [inline index.bs]: "sending the content" is a bit vague. Is this defined somewhere else?

File: index.bs [hunk 2]
```diff
@@ -187,7 +188,27 @@ Presentation API Requirements {#requirements-presentation-api}
...
+    {{HTMLMediaElement}} to a compatible remote playback device.
+
+3.  The {{HTMLMediaElement}} content to be sent should include both media data
+    and text tracks.
+
+4.  During remote playback, the controlling user agent and the remote playback
```
Comment (mounirlamouri) [inline index.bs]: Should we define what controlling means here?

File: index.bs (hunk 1 above)
Comment (mfoltzgoogle) [reply index.bs]: I rephrased this to use the term _initiate remote playback_ from the Remote Playback spec which seems to be the closest in meaning.

File: index.bs (hunk 2 above)
Comment (mfoltzgoogle) [reply index.bs]: "controlling user agent" is a term defined in this spec, but inherited from the Presentation API.   The Remote Playback requirements use the generic term "user agent" which I can also use here if you prefer.

We have a little work to do to reconcile the terminology between the three specs (Issue #144) :-)

**Comment by mfoltzgoogle** [comment]
Thanks for the tip on Travis.  It hasn't been a big issue to manually regenerate files so far, but as we have more contributors it would be helpful to add automation in the future.

**Comment by anssiko** [comment]
Thanks for the tip @mounirlamouri

I drive-by installed pr-preview integration for `webscreens/openscreenprotocol` and added a [config file](https://github.com/webscreens/openscreenprotocol/blob/gh-pages/.pr-preview.json) with default settings. Now future PRs should have a Preview and Diff links in them.

(See https://github.com/tobie/pr-preview for more information.)

File: index.bs [hunk 3]
```diff
@@ -193,7 +195,27 @@ Presentation API Requirements {#requirements-presentation-api}
...
 
-Issue(3): Requirements for Remote Playback API.
+1.  A [=controlling user agent=] must be able to find out whether there is at
+    least one compatible [=remote playback device=] available for a given
+    {{HTMLMediaElement}}, both instantaneously and continuously.
+
+2.  A controlling user agent must be able to to [=initiate remote playback=] of
+    an {{HTMLMediaElement}} to a compatible remote playback device.
+
+3.  The controlling user agent must be able send media data and text tracks
```
Comment (pthatcherg) [inline index.bs]: Send media data?  Like for remoting?  Hmm... I have not included any messages like that so far.  Should I make a PR to do so?   I guess we could just use the streaming messages that landed recently.

Comment (mfoltzgoogle) [reply index.bs]: I wasn't intending to suggest we were tackling remoting just yet (i.e. real time streaming of data from a media element).  I'll update the language to specifically address the flinging case before merging.

**Comment by mfoltzgoogle** [comment]
Merged now; we'll review this at the F2F.

File: index.bs [hunk 4]
```diff
@@ -193,7 +195,27 @@ Presentation API Requirements {#requirements-presentation-api}
 Remote Playback API Requirements {#requirements-remote-playback}
 ----------------------------------------------------------------
 
-Issue(3): Requirements for Remote Playback API.
+1.  A [=controlling user agent=] must be able to find out whether there is at
+    least one compatible [=remote playback device=] available for a given
+    {{HTMLMediaElement}}, both instantaneously and continuously.
+
+2.  A controlling user agent must be able to to [=initiate remote playback=] of
```
Comment (tidoust) [inline index.bs]: s/to to/to/

**Pull request merged**